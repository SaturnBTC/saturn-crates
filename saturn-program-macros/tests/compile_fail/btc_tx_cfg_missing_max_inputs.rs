use saturn_program_macros::saturn_program;

mod instruction {
    #[derive(borsh::BorshSerialize, borsh::BorshDeserialize)]
    pub enum Instr {
        Dummy,
    }
}

#[saturn_program(
    btc_tx_cfg(max_modified_accounts = 2) // missing max_inputs_to_sign
)]
mod handlers {}

fn main() {} 