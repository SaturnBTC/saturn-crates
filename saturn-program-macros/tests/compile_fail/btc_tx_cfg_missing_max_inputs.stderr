error: `btc_tx_cfg` must specify `max_inputs_to_sign`
  --> tests/compile_fail/btc_tx_cfg_missing_max_inputs.rs:10:1
   |
10 | / #[saturn_program(
11 | |     instruction = "crate::instruction::Instr",
12 | |     btc_tx_cfg(max_modified_accounts = 2) // missing max_inputs_to_sign
13 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `saturn_program` (in Nightly builds, run with -Z macro-backtrace for more info)
